### ISSUE TITLE
More detailed Logging

### ISSUE THREAD

**Issue by tb582** [rclone/rclone#7463]
Problem:

Currently, Rclone's debug logs indicate when files are excluded, but they do not specify the exact reason for the exclusion. This lack of detail makes it challenging for users to determine whether a file is excluded due to path filters, max-age filters, or other reasons. This can lead to difficulties in troubleshooting and understanding Rclone's behavior, especially in complex synchronization setups.
Proposed Feature:

I propose enhancing Rclone's logging capabilities to include detailed reasons for the exclusion of files and directories. Specifically, the debug logs should explicitly state whether a file is excluded due to not matching include rules, being caught by exclude rules, or not meeting criteria such as max-age.
Implementation Suggestion:

When a file or directory is excluded, the log entry should include a tag or a clear message stating the specific filter or condition leading to its exclusion. For example:

    Excluded (Path Filter): [file/directory path]
    Excluded (Max-Age Filter): [file/directory path]
    Excluded (Size Filter): [file/directory path]

This enhancement would involve modifying the logging system to track the reason for each file's exclusion and include this information in the debug output.
Benefits:

    Improved Troubleshooting: Users will be able to quickly identify why a particular file was excluded, making it easier to debug and adjust their Rclone configurations.
    Enhanced Transparency: Providing more detailed logs will increase transparency in how Rclone processes files according to user-defined rules.
    Time-Saving: This feature will save users time by reducing the need for trial and error in understanding Rclone's behavior.

I believe this enhancement will significantly benefit all Rclone users, especially those dealing with complex synchronization tasks and detailed filtering rules.

**Comment by tb582** [comment]
@ncw  you think this is doable ? :)

**Comment by ncw** [comment]
The root of all filtering is this function

https://github.com/rclone/rclone/blob/2f5685b40562cf764da673543245ddb13c049a4a/fs/filter/filter.go#L332-L334

We'd need to adjust it to return a `reason string` and keep passing that reason up to the places where it is logged.

We'd need to add logging to `fs/march` for excluded files. We have this in `fs/walk` but not in `fs/march` for some reason.

```
fs/walk/walk.go:					fs.Debugf(entry, "Excluded from sync (and deletion)")
fs/walk/walk.go:					fs.Debugf(x, "Excluded from sync (and deletion)")
fs/walk/walk.go:							fs.Debugf(basename, "Excluded from sync (and deletion) based on exclude file")
fs/walk/walk.go:					fs.Debugf(x, "Excluded from sync (and deletion)")
```

Or maybe put all the logging in the `filter.Include` function - I wonder if that would work and be sensible.

**Comment by KDreynolds** [comment]
If no one minds I can take this and try the solutions mentioned by @ncw above

**Comment by tb582** [comment]
Hi @KDreynolds any updates here ?

**Comment by KDreynolds** [comment]
Been tied up a little bit with some other stuff, should be able to look into this today and hopefully get something in very soon

**Comment by KDreynolds** [comment]
just pushed some updates on a separate project, so i can actually get this done here soon.

**Comment by KDreynolds** [comment]
https://github.com/rclone/rclone/pull/7546

pull request is in.

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.66

Thank you @KDreynolds
